pyiron_base/generic/datacontainer.py

```diff
@@ -811,8 +811,14 @@ def _from_hdf(self, hdf, version=None):
             def normalize_key(name):
                 # split a dataset/group name into the position in the list and
                 # the key
-                k, i = name.split("__index_", maxsplit=1)
-                i = int(i)
+                try:
+                    k, i = name.split("__index_", maxsplit=1)
+                    i = int(i)
+                except ValueError:
```

Comment:
Do we need the `try except` or can we split into a list and then check the size of the list?

Reply:
We can do that, but we also need to check whether `i` can be parsed as an `int`. Personally I'd think at that point checking explicitly is more convoluted.  In the common case where no exception is thrown `try`/`expect` as supposedly faster also, though I think it hardly matters here.  Anyways, I don't feel strongly about this and can change it.